### PR TITLE
refactor(router/v1): remove unused fields

### PR DESCRIPTION
**Pull request by domodwyer**
I noticed these are never going to be used (I think?)

---

* refactor(v1): remove unused fields (3ba01f5b0)
      
      We do not intend to do anything with these values, so we can skip
      deserialising them, and therefore remove their types too!
      
      Serde does not perform "strict" deserialisation by default (it does not
      error when it sees a field it does not know about). You can enable
      "strict" mode by adding an #[serde(deny_unknown_fields)] annotation:
      
          https://serde.rs/container-attrs.html#deny_unknown_fields

### CODE REVIEW

File: router/src/server/http/write_v1.rs [hunk 1]
```diff
@@ -75,21 +57,14 @@ impl Default for RetentionPolicy {
...
-#[derive(Debug, Deserialize)]
 /// Query Parameters for v1 DML operation.
+#[derive(Debug, Deserialize)]
 pub(crate) struct WriteParamsV1 {
     pub(crate) db: String,
 
-    #[allow(dead_code)]
-    u: Option<String>,
-    #[allow(dead_code)]
-    p: Option<String>,
```
Comment (wiedld) [inline router/src/server/http/write_v1.rs:86]: I have auth code locally, which consumes this `p` field. As per the authz/authn spec laid out in [this ticket](https://github.com/influxdata/idpe/issues/17265).

File: ioxd_router/src/lib.rs [hunk 2]
```diff
@@ -474,7 +474,7 @@ pub async fn create_router2_server_type(
 // NOTE!!! This needs to be kept in sync with `create_router2_server_type` until the
 // switch to the RPC write path/ingester2 is complete! See the numbered sections that annotate
 // where these two functions line up and where they diverge.
-pub async fn create_router_server_type<'a>(
+pub async fn create_router_server_type(
```
Comment (wiedld) [inline ioxd_router/src/lib.rs:477]: So clippy doesn't catch an unused lifetime?  😭

**Review by wiedld — CHANGES_REQUESTED** [review]
Please add the `p` field back in. Then this is gtg.

File: router/src/server/http/write_v1.rs [hunk 3]
```diff
@@ -75,21 +57,14 @@ impl Default for RetentionPolicy {
...
-
-    #[allow(dead_code)]
-    #[serde(default)]
-    consistency: Consistency,
```
Comment (wiedld) [inline router/src/server/http/write_v1.rs:90]: Also, you will need to remove the `invalid_consistency` test.

**Comment by domodwyer** [comment]
Superseded by https://github.com/influxdata/influxdb_iox/pull/7429

**Pull request closed without merging**